pkg/objectstate/merge/merge.go

```diff
@@ -148,3 +161,20 @@ func preserveServiceAccountPullSecrets(original, mutated *corev1.ServiceAccount)
 
 	mutated.Secrets = original.Secrets
 }
+
+// preserve the IP configuration from the original object since
+// those are assigned by external operator (not ours)
+func preserveIPConfigurations(original, mutated *corev1.ServiceSpec) {
```

Comment:
makes sense. I guess we will need to play catchup to find cases like this in the future?
nit: please make the comment docstring compliant:
```
// preserveIPConfigurations keep the configuration from the original object
// because it is owned by an external operator (not ours)
```